src/bgw/job.c

```diff
@@ -1315,18 +1316,18 @@ ts_bgw_job_run_and_set_next_start(BgwJob *job, job_main_func func, int64 initial
 								  Interval *next_interval, bool atomic, bool mark)
 {
 	BgwJobStat *job_stat;
-	bool had_error;
+	bool result;
 
 	if (atomic)
 		StartTransactionCommand();
 
 	if (mark)
 		ts_bgw_job_stat_mark_start(job->fd.id);
 
-	had_error = func();
+	result = func();
 
 	if (mark)
-		ts_bgw_job_stat_mark_end(job, had_error ? JOB_FAILURE : JOB_SUCCESS);
+		ts_bgw_job_stat_mark_end(job, result ? JOB_SUCCESS : JOB_FAILURE);
```

Comment:
I find this very puzzling...there is no contract for what implementations of `job_main_func` supposed to live up to....

...and now this commit is just negating that...seems like `ts_telemetry_main` is the only implementation ? :facepalm:

Reply:
Yes, but don't want to make larger changes than necessary since things have a tendency to break. Fine with refactoring code as a separate PR though.